### PR TITLE
Assign courses via API

### PR DESCRIPTION
**Pull request by Bjvanminnen**
The next step in getting to the point where we can assign courses via teacher-dashboard.

This PR adds logic to the create/update methods of DashboardSection to be able to set a course_id. The UI still exposes no way to use this, but I've added some test coverage.

### CODE REVIEW

File: pegasus/test/test_section_api_helpers.rb [hunk 1]
```diff
@@ -278,30 +282,222 @@ class SectionApiHelperTest < SequelTestCase
...
+          row_id = DashboardSection.create(params)
+
+          row = Dashboard.db[:sections].where(id: row_id).first
+          assert_equal 15, row[:user_id]
+          assert_equal 'New Section', row[:name]
+          assert_equal 'word', row[:login_type]
+          assert_nil row[:script_id]
+          assert_nil row[:course_id]
+          assert_nil row[:grade]
+          assert !row[:code].nil?
```
Comment (ashercodeorg) [inline pegasus/test/test_section_api_helpers.rb:309]: nit: Prefer `refute_nil row[:code]`. Also below.

File: pegasus/test/test_section_api_helpers.rb [hunk 2]
```diff
@@ -278,30 +282,222 @@ class SectionApiHelperTest < SequelTestCase
...
+          assert_equal false, row[:stage_extras]
```
Comment (ashercodeorg) [inline pegasus/test/test_section_api_helpers.rb:310]: nit: Prefer `refute row[:stage_extras]`, assuming falsiness is sufficient. Also below.

File: pegasus/test/fixtures/fake_dashboard.rb [hunk 3]
```diff
@@ -73,6 +73,29 @@ module FakeDashboard
...
+  #
+  # Fake Data: Scripts
+  #
+  SCRIPTS = [
+    {
+      id: 10,
+      name: 'flappy',
+      hidden: 0
+    }
+  ]
```
Comment (ashercodeorg) [inline pegasus/test/fixtures/fake_dashboard.rb:96]: Though it is great to see the `scripts` table being seeded, I'm not seeing usage of this in the tests. Is that intentional?

**Comment by ashercodeorg** [comment]
It is probably worth confirming manaully that the teacher dashboard still functions normally, be it locally or on staging (post-merge).

File: pegasus/test/fixtures/fake_dashboard.rb (hunk 3 above)
Comment (Bjvanminnen) [reply pegasus/test/fixtures/fake_dashboard.rb:96]: When assigning a script to a section, we check to see that it's a valid one. If the script does not exist in our db, it's considered invalid (thus tests will fail without this script in our fake db).

File: pegasus/test/test_section_api_helpers.rb (hunk 2 above)
Comment (Bjvanminnen) [reply pegasus/test/test_section_api_helpers.rb:310]: The schema appears to allow for both null and false as 'falsey' values. I'm not sure we actually expect/can handle both, and feel better about having the more strict check that this value is false

**Comment by Bjvanminnen** [comment]
Validated locally that script assignment still works (via the UI). Also clicked around through my teacher dashboard some, and couldn't find anything amiss.

**Pull request merged**